src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2731,27 +2731,26 @@ private string ReCreateParts(UriComponents parts, ushort nonCanonical, UriFormat
                 {
                     mode = UnescapeMode.CopyOnly;
                 }
+
                 // NormalizedHost
                 if ((parts & UriComponents.NormalizedHost) != 0)
                 {
-                    unsafe
+                    stemp = UriHelper.StripBidiControlCharacters(stemp, stemp);
+
+                    var hostBuilder = new ValueStringBuilder(stackalloc char[512]);
```

Comment:
512 is chosen as that is the value of the `StackallocThreshold` constant being introduced in #34864.